pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancerClusterStatelessTest.java

```diff
@@ -246,19 +246,19 @@ public void testRebalance()
     tableConfig.setInstanceAssignmentConfigMap(null);
     _helixResourceManager.updateTableConfig(tableConfig);
 
-    // Without instances reassignment, the rebalance should return status NO_OP as instance partitions are already
-    // generated
+    // Without instances reassignment, the rebalance should return status DONE,
```

Comment:
This doesn't seem correct. Without instance reassignment, we should not remove the instance partitions, and it should return NO_OP. Instance partitions should be removed when instance reassignment is enabled

Reply:
https://github.com/apache/pinot/blob/master/pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java#L559-L572

This code here will remove instance partitions whether instance reassignment or not.

Reply:
Yes, you are right. Ideally we should only modify/remove instance partitions when instance reassignment is enabled, but currently that is not the case.
We can fix this in a separate PR because this is not introduced in this PR, and it also affects other classes

Reply:
Filed #11169 to fix this